### DISCUSSION TITLE
Extend JWT token expiration time

### DISCUSSION THREAD

**Discussion by sagilefl** [Web3Auth/Web3Auth#865] [Help]
I noticed that the JWT tokens received for social login via Google, Twitter or Discord are valid for only 24 hours. 
Is there a way to extend the expiration time, or use a refresh token to retrieve a new valid JWT token after it expires?

**Comment by rinzler99** [comment]
Hey can you please refer to this docs: https://web3auth.io/docs/server-side-verification/social-login-users#sample-idtoken-payload
You can use "exp" param to at the jwt expiration time as below:
<img width="1308" alt="image" src="https://user-images.githubusercontent.com/30899350/201528403-642c7e76-5710-4e3c-9249-95e8ebf00961.png">

**Comment by himanshuchawla009** [comment]
Hey, you can use your own social logins by creating a custom verifier to have your own custom configuration for social logins. Please refer to this doc to do so: https://web3auth.io/docs/developer-dashboard/setup-custom-authentication
